_docs/src/documentation/test_projects.md

```diff
@@ -0,0 +1,348 @@
+
+TEST_CASE("My Test Case")
+{
+    // Test Code Here
+}
+```
+
+### Auto Test Project
+An auto test is ideal to test a complex system or test different systems working together. It'll launch an app, backed by the rex engine, in such a way that no user input is needed.
+
+Just like before, let's work from an example. Let's pretend we want to create a auto test project for the editor called `ReginaAutoTests`
+To create a auto test project, create a new folder in ~/tests under the specific subfolder you need, in this case, because we're making an auto test for the editor, we'll make a folder under `4_tools`.
+
+Like any other project, we need to create a `src` and `include` folder and a `sharpmake file`.
+
+this would give us the following folder structure
+- ~/tests/4_tools/regina_auto_tests/include
+- ~/tests/4_tools/regina_auto_tests/src
+- ~/tests/4_tools/regina_auto_tests/regina_auto_tests.sharpmake.cs
+
+if needed, you also need to create a code generation config file under `~/config/4_tools/regina_auto_tests`. For this example, we don't need such a file.
+
+Next you need to configure the sharpmake project. You do this like you'd configure any other project, but auto unit test project needs 2 things to work properly.
+- The `ProjectType` needs to be specified in the constructor. Because we're generating a auto test, specify `ProjectType = ProjectGen.TestProjectType.AutoTest;`.
+- We need to add a dependency to `RexAutoTest` project. Override the `SetupLibDependencies` func and add a public dependency to `RexAutoTest`.
+
+An example of such a file could look like this
+```cs
+using Sharpmake;
+using System.IO;
+
+[Generate]
+public class ReginaAutoTests : TestProject
+{
+  public ReginaAutoTests() : base()
+  {
+    Name = GenerateName("ReginaAutoTests");
+    GenerateTargets();
+
+    string ThisFileFolder = Path.GetDirectoryName(Utils.CurrentFile());
+    SourceRootPath = ThisFileFolder;
+
+    ProjectType = ProjectGen.TestProjectType.AutoTest;
+  }
+
+  protected override void SetupSolutionFolder(RexConfiguration conf, RexTarget target)
+  {
+    base.SetupSolutionFolder(conf, target);
+
+    conf.SolutionFolder = Path.Combine(conf.SolutionFolder, "4_tools");
+  }
+
+  protected override void SetupLibDependencies(RexConfiguration conf, RexTarget target)
+  {
+    base.SetupLibDependencies(conf, target);
+
+    conf.AddPublicDependency<Regina>(target, DependencySetting.Default | DependencySetting.IncludeHeadersForClangtools);
+    conf.AddPublicDependency<RexAutoTest>(target, DependencySetting.Default);
+  }
+
+  protected override void SetupConfigSettings(RexConfiguration conf, RexTarget target)
+  {
+    base.SetupConfigSettings(conf, target);
+
+    conf.Options.Remove(Options.Vc.Compiler.JumboBuild.Enable);
+  }
+}
+```
+
+Next you need to create a source file which will act as the entry point of the auto tests, let's call this file `regina_auto_tests.cpp`
+Fill the file with the following code
+
+```cpp
```

Comment:
This seems like a lot more boiler plate isn't there any way to make this more user friendly I suppose most of the files would look quite similar. Isn't it possible to create a separate file that exposes the functions a user need and all the includes and basic setup are just within some other header file?